### ISSUE TITLE
Transfer learning unable to do 

### ISSUE THREAD

**Issue by ravitejarj** [zhanghang1989/PyTorch-Encoding#391]
Hi @zhanghang1989 
I am unable to do transfer learning on the model
i have downloaded **get_deeplab_resnest101_ade and**
when i changed the no of classes in **ade20k.py** (no of classes 8)
pre trained model is not loading ( getting error )

So I have changed the code for Transfer learning
code changes:
1) deeplab.py
in get_deeplab_resnest101_ade function changed from
from
#model = DeepLabV3(datasets[dataset.lower()].NUM_CLASS, backbone=backbone, root=root, **kwargs)
to 
no_of_classes = 150
model = DeepLabV3(no_of_classes, backbone=backbone, root=root, **kwargs)

so i can load pretrained model with 150 classes
then
2) In train_dist.py file
# Model loading
model = get_deeplab_resnest101_ade(pretrained=True)
for param in model.parameters():
    param.requires_grad = False
model.auxlayer = Sequential((Conv2d(1024, 256, kernel_size=(3, 3), stride=(1, 1),padding=(1, 1), bias=False)),
                            (BatchNorm2d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)),
                            (ReLU()),
                            (Dropout(p=0.1, inplace=False)),
                            (Conv2d(256, 8, kernel_size=(1, 1), stride=(1, 1))))
for param in model.auxlayer.parameters():
    param.requires_grad = True

**Training**
python train_dist.py --dataset ade20k --model deeplab --aux --backbone resnest101 --ft --epochs 100

after successful training
i am getting 150 classes output not 8 classes(i have given 8 classes in last layer)

can you help me what to do

**Comment by zhanghang1989** [comment]
An easy solution is set `strict=False` when loading the pretrained model
https://pytorch.org/docs/master/generated/torch.nn.Module.html?highlight=load_state_dict#torch.nn.Module.load_state_dict

**Comment by ravitejarj** [comment]
i have changed code in **deeplab.py**
model.load_state_dict(torch.load(
            get_model_file('deeplab_%s_%s'%(backbone, acronyms[dataset]),root=root)), strict=False)

then trained.
### Inference code model loading
model = get_deeplab_resnest101_ade(pretrained = False)
model.auxlayer = Sequential((Conv2d(1024, 256, kernel_size=(3, 3), stride=(1, 1),padding=(1, 1), bias=False)),
                            (BatchNorm2d(256, eps=1e-05, momentum=0.1, affine=True, track_running_stats=True)),
                            (ReLU()),
                            (Dropout(p=0.1, inplace=False)),
                            (Conv2d(256, 8, kernel_size=(1, 1), stride=(1, 1))))
checkpoint = torch.load(model_path)
model.load_state_dict(checkpoint['state_dict'])
model.eval()
model.cpu()
print(">> Model loaded...")

**for training model i have uploaded above**

these are the changes i have made in code.

after that still i am getting 150 class output.

is this correct ?

**Comment by ravitejarj** [comment]
@zhanghang1989 
IN **deeplabv3** i can change output channels by
model.classifier = DeepLabHead(2048, outputchannels)

but in this code(Pytorch Encoding Deeplab) how can i changed output channels
can you please help me

**Comment by ravitejarj** [comment]
Thank you
